src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java

```diff
@@ -0,0 +1,241 @@
+import org.opentripplanner.middleware.models.MonitoredTrip;
+import org.opentripplanner.middleware.otp.OtpDispatcherResponse;
+import org.opentripplanner.middleware.otp.OtpRequest;
+import org.opentripplanner.middleware.otp.response.Itinerary;
+import org.opentripplanner.middleware.otp.response.OtpResponse;
+import org.opentripplanner.middleware.otp.response.Place;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.opentripplanner.middleware.TestUtils.TEST_RESOURCE_PATH;
+import static org.opentripplanner.middleware.otp.OtpDispatcherResponseTest.DEFAULT_PLAN_URI;
+import static org.opentripplanner.middleware.utils.DateTimeUtils.DEFAULT_DATE_FORMAT_PATTERN;
+import static org.opentripplanner.middleware.utils.ItineraryUtils.DATE_PARAM;
+import static org.opentripplanner.middleware.utils.ItineraryUtils.IGNORE_REALTIME_UPDATES_PARAM;
+
+public class ItineraryUtilsTest extends OtpMiddlewareTest {
+    private static final Logger LOG = LoggerFactory.getLogger(ItineraryUtilsTest.class);
+    /** Abbreviated query for the tests */
+    public static final String BASE_QUERY = "?fromPlace=2418%20Dade%20Ave&toPlace=McDonald%27s&date=2020-08-13&time=11%3A23&arriveBy=false";
+
+    // Date and time from the above query.
+    public static final String QUERY_DATE = "2020-08-13";
+    public static final String QUERY_TIME = "11:23";
+
+    private static OtpDispatcherResponse otpDispatcherPlanResponse;
+    private static OtpDispatcherResponse otpDispatcherPlanErrorResponse;
+
+    @BeforeAll
+    public static void setup() throws IOException {
+        TestUtils.mockOtpServer();
+
+        // Contains an OTP response with an itinerary found.
+        // (We are reusing an existing response. The exact contents of the response does not matter
+        // for the purposes of this class.)
+        String mockPlanResponse = FileUtils.getFileContents(
+            TEST_RESOURCE_PATH + "persistence/planResponse.json"
+        );
+        // Contains an OTP response with no itinerary found.
+        String mockErrorResponse = FileUtils.getFileContents(
+            TEST_RESOURCE_PATH + "persistence/planErrorResponse.json"
+        );
+
+        otpDispatcherPlanResponse = new OtpDispatcherResponse(mockPlanResponse, DEFAULT_PLAN_URI);
+        otpDispatcherPlanErrorResponse = new OtpDispatcherResponse(mockErrorResponse, DEFAULT_PLAN_URI);
+    }
+
+    @AfterEach
+    public void tearDownAfterTest() {
+        TestUtils.resetOtpMocks();
+    }
+
+    /**
+     * Test case in which all itineraries exist and result.allCheckedDatesAreValid should be true.
+     */
+    @Test
+    public void testAllItinerariesExist() throws URISyntaxException {
```

Comment:
Just a minor comment on test names: we don't usually call them `testMethodName`. I would prefer all these test names match our other syntax. E.g., `canCheckValidItineraries` or something.

Reply:
Updated in https://github.com/ibi-group/otp-middleware/pull/77/commits/a655afcdf5b8e76d68cfecf0e6ba3edb2a5d43fa.